src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,54 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.stream.Collectors;
+
 public class SalaryInfo {
+    private static final String FIRST_LINE_FORMAT = "Report for period %s - %s";
+    private static final String SALARY_LINE_FORMAT = "\n%s - %d";
+    private static final int DEFAULT_SALARY_VALUE = 0;
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        LocalDate dateFromLocal = LocalDate.parse(dateFrom, SalaryData.DATE_TIME_FORMATTER);
+        LocalDate dateToLocal = LocalDate.parse(dateTo, SalaryData.DATE_TIME_FORMATTER);
+        DateRange dateRange = new DateRange(dateFromLocal, dateToLocal);
+        Map<String, Integer> nameSalaryMap = getNameSalaryMap(names, data, dateRange);
```

Comment:
Don’t create redundant loops. Two is enough. Here, using streams and then looping over the names array constitutes more than two loops when considering the internal implementation of streams. Try to refactor the code to use at most two loops.